java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,250 @@
+     */
+    @RequiredArgsConstructor
+    @Getter
+    public static class RangeByIndex implements IScoredRangeQuery {
+        /** The start index of the range. */
+        private final String start;
+
+        /** The stop index of the range. */
+        private final String end;
+
+        public RangeByIndex(long start, long end) {
+            this.start = Long.toString(start);
+            this.end = Long.toString(end);
+        }
+
+        @Override
+        public Limit getLimit() {
+            return null;
+        }
+    }
+
+    /**
+     * Represents a range by score in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent score boundaries.
+     */
+    @Getter
+    public static class RangeByScore implements IScoredRangeQuery {
+        /** The start score boundary. */
+        private final String start;
+
+        /** The stop score boundary. */
+        private final String end;
+
+        /**
+         * The limit argument for a range query. Defaults to null. See <code>Limit</code> class for more
+         * information.
+         */
+        private final Limit limit;
+
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start,
+                @NonNull RangeOptions.ScoreRange end,
+                @NonNull Limit limit) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = limit;
+        }
+
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start, @NonNull RangeOptions.ScoreRange end) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = null;
+        }
+    }
+
+    /**
+     * Represents a range by lexicographical order in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent lexicographical boundaries.
+     */
+    @Getter
+    public static class RangeByLex implements IRangeQuery {
+        /** The start lexicographic boundary. */
+        private final String start;
+
+        /** The stop lexicographic boundary. */
+        private final String end;
+
+        /**
+         * The limit argument for a range query. Defaults to null. See <code>Limit</code> class for more
+         * information.
+         */
+        private final Limit limit;
+
+        public RangeByLex(
+                @NonNull RangeOptions.LexRange start,
+                @NonNull RangeOptions.LexRange end,
+                @NonNull Limit limit) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = limit;
+        }
+
+        public RangeByLex(@NonNull RangeOptions.LexRange start, @NonNull RangeOptions.LexRange end) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = null;
+        }
+    }
+
+    public static String[] createZrangeArgs(
```

Comment:
We will have two createZrangeArgs calls if we move `IScoredRangeQuery`.  Might look into re-writing this so that we don't have too much shared code.